Fail
file path is :
projects/deltachat-core/rust/src/context.rs

output is:


error is :
   Compiling deltachat v1.139.6 (/home/k/ogs_PHD/LLM4SE/2Rust_projects/auto_get_functionPair/projects/deltachat-core/rust)
error[E0425]: cannot find value `DC_CONTEXT_MAGIC` in this scope
    --> src/context.rs:1154:22
     |
1154 |     if self.magic != DC_CONTEXT_MAGIC {
     |                      ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_STATE_IN_FRESH` in this scope
    --> src/context.rs:1172:18
     |
1172 |     stmt.bind(1, DC_STATE_IN_FRESH)?;
     |                  ^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CHAT_ID_LAST_SPECIAL` in this scope
    --> src/context.rs:1173:18
     |
1173 |     stmt.bind(2, DC_CHAT_ID_LAST_SPECIAL)?;
     |                  ^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
     |
help: consider importing this constant
     |
3    + use crate::constants::DC_CHAT_ID_LAST_SPECIAL;
     |

error[E0425]: cannot find value `DC_CHAT_DEADDROP_BLOCKED` in this scope
    --> src/context.rs:1174:37
     |
1174 |     stmt.bind(3, if show_deaddrop { DC_CHAT_DEADDROP_BLOCKED } else { 0 })?;
     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `NO_PARAMS` in this scope
    --> src/context.rs:1176:40
     |
1176 |     while let Ok(row) = stmt.query_row(NO_PARAMS, |row| row.get(0)).await {
     |                                        ^^^^^^^^^ not found in this scope

warning: unused import: `MessageState`
  --> src/context.rs:31:37
   |
31 | use crate::message::{self, Message, MessageState, MsgId, Viewtype};
   |                                     ^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> src/lib.rs:6:5
   |
6  |     unused,
   |     ^^^^^^
   = note: `#[warn(unused_imports)]` implied by `#[warn(unused)]`

error[E0609]: no field `magic` on type `&context::Context`
    --> src/context.rs:1154:13
     |
1154 |     if self.magic != DC_CONTEXT_MAGIC {
     |             ^^^^^ unknown field
     |
     = note: available field is: `inner`
     = note: available fields are: `blobdir`, `sql`, `smeared_timestamp`, `running_state`, `generating_key_mutex` ... and 19 others

error[E0599]: no method named `prepare` found for struct `Sql` in the current scope
    --> src/context.rs:1158:25
     |
1158 |     let stmt = self.sql.prepare(
     |                ---------^^^^^^^ method not found in `Sql`
     |
    ::: src/sql.rs:57:1
     |
57   | pub struct Sql {
     | -------------- method `prepare` not found for this struct
     |
     = help: items from traits can only be used if the trait is implemented and in scope
     = note: the following traits define an item `prepare`, perhaps you need to implement one of them:
             candidate #1: `precis_core::profile::Profile`
             candidate #2: `precis_core::profile::PrecisFastInvocation`

error[E0599]: no method named `prepare` found for struct `sql::Sql` in the current scope
    --> src/context.rs:1158:25
     |
1158 |     let stmt = self.sql.prepare(
     |                ---------^^^^^^^ method not found in `sql::Sql`
     |
    ::: src/sql.rs:57:1
     |
57   | pub struct Sql {
     | -------------- method `prepare` not found for this struct
     |
     = help: items from traits can only be used if the trait is implemented and in scope
     = note: the following traits define an item `prepare`, perhaps you need to implement one of them:
             candidate #1: `precis_core::profile::Profile`
             candidate #2: `precis_core::profile::PrecisFastInvocation`

Some errors have detailed explanations: E0425, E0599, E0609.
For more information about an error, try `rustc --explain E0425`.
warning: `deltachat` (lib) generated 1 warning
error: could not compile `deltachat` (lib) due to 7 previous errors; 1 warning emitted
warning: build failed, waiting for other jobs to finish...
warning: `deltachat` (lib test) generated 1 warning (1 duplicate)
error: could not compile `deltachat` (lib test) due to 7 previous errors; 1 warning emitted
error: command `/home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics` exited with code 101

